src/components/Banner/BannerContextManager.js

```diff
@@ -25,7 +36,28 @@ let activeBanner = null;
 
 if (activeBanner) {
   try {
-    if (localStorage.getItem(activeBanner.storageId)) {
+    if (localStorage[activeBanner.storageId]) {
+      snooze_start_date = new Date(
+        parseInt(localStorage.getItem(activeBanner.storageId), 10),
+      );
+    }
+  } catch (err) {
+    // Ignore.
+  }
+
+  try {
+    // If it's too early or long past the campaign, don't show the banner:
+    if (
+      today < new Date(activeBanner.campaign_start_date) ||
+      today > new Date(activeBanner.campaign_end_date)
+    ) {
+      activeBanner = null;
+      // If we're in the campaign window, but the snooze has been set and it hasn't expired:
+    } else if (
+      snooze_start_date &&
+      addTimes(snooze_start_date, activeBanner.snooze_duration) >=
+        Date.parse(today.toString())
+    ) {
```

Comment:
Nit: I think `today` is already a date here:

```suggestion
    } else if (
      snooze_start_date &&
      addTimes(snooze_start_date, activeBanner.snooze_duration) >= today
    ) {
```